api/helpers/gamestate/moves/face-card/validate.js

```diff
@@ -0,0 +1,62 @@
+const GamePhase = require('../../../../../utils/GamePhase.json');
+
+module.exports = {
+  friendlyName: 'Validate request to play face card',
+
+  description: 'Verifies whether a request to play face card is legal, throwing explanatory error if not.',
+
+  inputs: {
+    currentState: {
+      type: 'ref',
+      descriptions: 'Object containing the current game state',
+      required: true,
+    },
+    /**
+     * @param { Object } requestedMove - Object describing the request to play face card
+     * @param { String } requestedMove.cardId - Card Played as face card
+     * @param { MoveType.FACECARD } requestedMove.moveType - Specifies that this a face card move
+     */
+    requestedMove: {
+      type: 'ref',
+      description: 'Object containing data needed for current move',
+      required: true,
+    },
+    playedBy: {
+      type: 'number',
+      description: 'Player number of player requesting move',
+      required: true,
+    },
+  },
+  sync: true,
+  fn: ({ requestedMove, currentState, playedBy }, exits) => {
+    try {
+      const playerKey = playedBy ? 'p1' : 'p0';
+
+      const cardPlayed = currentState[playerKey].hand.find(({ id }) => id === requestedMove.cardId);
```

Comment:
Last request: let's switch this to using the same ternary expression as the execute() helper so it gets the player object with currentState.p0 and currentState.p1 into a variable called `player`. That will be more consistent